tests/e2e/README.md

```diff
@@ -0,0 +1,118 @@
+
+### `upgrade` Package
+
+The `e2e` package defines an upgrade `Manager` abstraction. Suite will utilize `Manager`'s functions to run different versions of evmos containers, propose, vote and delegate.
+
+* `manager.go`: defines core manager logic for running containers, export state and create networks.
+
+* `govexec.go`: defines `gov-specific` exec commands to submit/delegate/vote thru nodes `gov` module.
+
+## Chain Upgrades
+
+e2e testing logic utilizes three parameters:
+
+```shell
+# INITIAL_VERSION is the tag of evmos node which will be used to build initial validators containers
+# by default it gets previous git tag from current, e.g. if current tag is v9.1.0 it will get v9.0.0 from git
+INITIAL_VERSION := $(shell git describe --abbrev=0 --tags `git rev-list --tags --skip=1 --max-count=1`)
+
+# current latest tag
+TARGET_VERSION := $(shell git describe --abbrev=0 --tags `git rev-list --tags --max-count=1`)
+
+# flag to skip containers cleanup after upgrade
+# should be set true with make test-e2e command if you need access to the node after upgrade
+E2E_SKIP_CLEANUP := false
+```
+
+every flag can be set manually with make command:
+
+```shell
+make test-e2e E2E_SKIP_CLEANUP=true INITIAL_VERSION=v8.2.0
+```
+
+Testing a chain upgrade is a three step process:
+
+1. Build a initial node version docker container (e.g. `v9.0.0`)
+2. Build a docker image for the evmos target version(local repo by default) (e.g. `v9.1.0`)
+3. Run tests
+
+### Run upgrade tests
+
+The e2e test will first run a `INITIAL_VERSION` node.
+
+The node will submit, deposit and vote for an upgrade proposal for upgrading to the `TARGET_VERSION`.
+
+After block `50` is reached, the test suite exports `/.evmosd` folder from docker container to local `build/` and than purge the container.
+
+Suite will mount `TARGET_VERSION` node to local `build/` dir and start the node. Node will get upgrade information from `upgrade-info.json` and will execute the upgrade.
+
+### Version retireve
+
+`INITIAL_VERSION` and `TARGET_VERSION` retieved from git tags by default with folowing commands:
+
+```shell
+# INITIAL_VERSION
+git describe --abbrev=0 --tags `git rev-list --tags --skip=1 --max-count=1`
+
+# TARGET_VERSION
+git describe --abbrev=0 --tags `git rev-list --tags --max-count=1`
+```
+
+If `Makefile` command cannot get the tags by some reason(i.e. you have no tag for local branch and want to upgrade from specific version to local node etc), versions should be specified manually:
+
+```shell
+make test-e2e INITIAL_VERSION=<version> TARGET_VERSION=<version>
+```
+
+`TARGET_VERSION` used as a software upgrade version in proposal and must match the version in `upgrades` package.
+
+### Testing Results
+
+Running the e2e test make script, will output the test results for each testing file. In case of an sucessfull upgrade the script will output like `ok  	github.com/evmos/evmos/v9/tests/e2e	174.137s`.
```

Comment:
```suggestion
Running the e2e test make script will output the test results for each testing file. In case of a successful upgrade, the script will print the following output (example):

```shell
ok  	github.com/evmos/evmos/v9/tests/e2e	174.137s.
```
```